language/tools/move-cli/src/base/docgen.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::base::prove::{Prove, ProverOptions};
+use clap::*;
+use move_package::BuildConfig;
+use std::path::PathBuf;
+
+/// Generating javadoc style documentation for packages using the Move Prover
+#[derive(Parser)]
+#[clap(name = "docgen")]
+pub struct Docgen;
+
+impl Docgen {
+    /// Simply calling the Move Prover with "--docgen"
+    pub fn execute(self, path: Option<PathBuf>, config: BuildConfig) -> anyhow::Result<()> {
+        let options = Some(ProverOptions::Options(vec!["--docgen".to_string()]));
```

Comment:
I think there are a bunch of options for docgen; I feel like we should expose them in the CLI here. Thoughts?

Reply:
@tnowacki What do you think, how stable these options will be in the future? Could we introduce `--` in order to forward extra options and avoid reduplication?

Reply:
No idea on the stability of it. It seems the args are rather hand rolled. It would ben nice to get them in the same struct opt definition if possible. But not necessary for this PR.

I would be fine with just duplicating them all here in struct opt, and fixing it later. Feels a bit more user friendly for this command.